libs/perun/components/src/lib/settings-mailing-lists/mailing-lists.component.ts

```diff
@@ -82,6 +86,17 @@ export class MailingListsComponent implements OnInit, OnDestroy {
         });
       })
       .unsubscribe();
+    this.routingSubscription = this.router.events
+      .pipe(filter((event): event is NavigationStart => event instanceof NavigationStart))
+      .subscribe((event) => {
+        if (!event.url.startsWith(location.pathname)) {
+          // clear the query params when navigating away
+          this.location.replaceState(
+            location.pathname,
+            this.clearParamsFromCurrUrl(['vo', 'resource'])
+          );
+        }
+      });
```

Comment:
Is any of this really necessary? I tried the code only with the change to ngDestroy and it seemed to work correctly

Reply:
It is there so that the params will stay removed also when going back in the browser.